src/core/service.c

```diff
@@ -2758,6 +2760,63 @@ static void service_run_next_main(Service *s) {
         (void) service_set_main_pidref(s, TAKE_PIDREF(pidref), &s->main_command->exec_status.start_timestamp);
 }
 
+static bool service_needs_reload_extensions(Unit *u) {
+        Service *s = ASSERT_PTR(SERVICE(u));
+
+        /* For now, opportunistically reload as long as we have an extension,
+         * and do not try to actually figure out if we have a vpick'd extension
+         * or not. */
+        return s->exec_context.n_extension_images > 0 ||
+                        !strv_isempty(s->exec_context.extension_directories);
+}
+
+static int service_reload_extensions(Unit *u) {
+        _cleanup_free_ char *root_dir = NULL, *propagate_dir = NULL, *private_namespace_dir = NULL,
+                *incoming_dir = NULL, *hierarchy_env = NULL;
+
```

Comment:
maybe add a "TODO: do this asynchronously" to note that this will change. As mentioned I'm working on this for the other live mounting stuff, so no need to do this in this PR, as it would duplicate that work